### ISSUE TITLE
`{{ *recursive children* }}` has unpredictable results.

### ISSUE THREAD

**Issue by curtisblackwell** [statamic/cms#2712]
## Bug Description
`{{ *recursive children* }}` has unpredictable results.

## How to Reproduce
Given this code:

```handlebars
{{ nav:navbar }}
  <li
    {{ if !last }}
      class="my-class-list"
    {{ else }}
      class="my-other-class-list"
    {{ /if }}
  >
    {{ if !children }}
      <a href="{{ url }}">{{ title }}</a>
    {{ else }}
      <div x-data="{ open: false }">
        <button @click="open = !open">
          {{ title }}
          <div class="inline-block w-2">{{ partial:components/svg/dropdown_triangle }}</div>
        </button>
        <ul x-cloak :class="{ 'hidden': !open }">
          <li><a href="{{ url }}">{{ title }}</a></li>
          {{ *recursive children* }}
        </ul>
      </div>
    {{ /if }}
  </li>
{{ /nav:navbar }}
```

…the conttents of `{{ partial:components/svg/dropdown_triangle }}` is the only thing output by `{{ *recursive children* }}` (once for each child).

Partial contents:
```xml
<svg viewBox="0 0 200 100" xmlns="http://www.w3.org/2000/svg" class="fill-current">
  <path d="M100.446 100.845L0.446045 0.84491L200.446 0.844926L100.446 100.845Z"/>
</svg>
```

With the partial empty, it outputs:

```handlebars
<ul :class="{ 'hidden': !open }" class="">
  <li><a href="/services">Services</a></li>              
</ul>
```

If I inline the SVG, everything works as expected.

## Extra Detail
[Discord conversation with Jason](https://discord.com/channels/489818810157891584/489819906540568593/768561558317695016)

## Environment
Statamic 3.0.16 Pro
Laravel 7.28.3
PHP 7.4.7
aryehraber/statamic-captcha 1.0.3
doublethreedigital/duplicator 1.1.0
edalzell/mailchimp 2.0.6
highland/statamic-array-get 1.0.0
statamic/seo-pro 2.0.13

**Install method (choose one):**
- Fresh install from `statamic/statamic`

**Comment by jasonvarga** [comment]
I haven't had a chance to test this exact example, but there are some other `*recursive children*` related fixes coming in 2.0.21. If you could update this issue after upgrading and letting us know whether it still happens, that'd be great.

I don't _think_ it would fix this issue, but it would be a nice happy accident if it did.

**Comment by curtisblackwell** [comment]
@jasonvarga Just checked, and the issue still exists in 3.0.21.
